health-services/project/src/main/java/org/egov/project/repository/rowmapper/LocationCaptureRowMapper.java

```diff
@@ -0,0 +1,91 @@
+        this.objectMapper = objectMapper;
+    }
+
+    /**
+     * Map a single row of the ResultSet to a UserAction object.
+     *
+     * @param resultSet The ResultSet containing data from the database.
+     * @param rowNum    The number of the current row.
+     * @return A UserAction object populated with data from the current row of the ResultSet.
+     * @throws SQLException If there is an issue accessing the ResultSet data.
+     */
+    @Override
+    public UserAction mapRow(ResultSet resultSet, int rowNum) throws SQLException {
+
+        // Creating AuditDetails object with information from the ResultSet
+        AuditDetails auditDetails = AuditDetails.builder()
+                .createdBy(resultSet.getString("createdBy"))
+                .createdTime(resultSet.getLong("createdTime"))
+                .lastModifiedBy(resultSet.getString("lastModifiedBy"))
+                .lastModifiedTime(resultSet.getLong("lastModifiedTime"))
+                .build();
+
+        // Creating client-specific AuditDetails object with information from the ResultSet
+        AuditDetails clientAuditDetails = AuditDetails.builder()
+                .createdTime(resultSet.getLong("clientCreatedTime"))
+                .createdBy(resultSet.getString("clientCreatedBy"))
+                .lastModifiedTime(resultSet.getLong("clientLastModifiedTime"))
+                .lastModifiedBy(resultSet.getString("clientLastModifiedBy"))
+                .build();
+
+        UserAction locationCaptureUserAction;
+        try {
+            // Building the UserAction object with data from the ResultSet
+            locationCaptureUserAction = UserAction.builder()
+                    .id(resultSet.getString("id"))
+                    .tenantId(resultSet.getString("tenantId"))
+                    .clientReferenceId(resultSet.getString("clientReferenceId"))
+                    .projectId(resultSet.getString("projectId"))
+                    .latitude(resultSet.getDouble("latitude"))
+                    .longitude(resultSet.getDouble("longitude"))
+                    .locationAccuracy(resultSet.getDouble("locationAccuracy"))
+                    .boundaryCode(resultSet.getString("boundaryCode"))
+                    .action(TaskAction.fromValue(resultSet.getString("action")))
+                    .auditDetails(auditDetails)
+                    .clientAuditDetails(clientAuditDetails)
+                    .additionalFields(resultSet.getString("additionalDetails") == null ? null : objectMapper
+                            .readValue(resultSet.getString("additionalDetails"), AdditionalFields.class))
+                    .build();
+        } catch (JsonProcessingException e) {
+            // Throwing a RuntimeException if there's an error processing JSON
+            log.error("Error processing Additional detail JSON in Location capture UserAction ", e);
+            throw new CustomException("JSON_PROCESSING_ERROR", "Error processing JSON: " + e.getMessage());
```

Comment:
This is not fine, error should have full key module_flow_errorreason like HOUHSEHOLD_SEARCH_LOCALITY_INVALID_ERROR like this

Reply:
done